scripts/update-tezos.sh

```diff
@@ -24,7 +24,7 @@ cd ..
 rm -rf upstream-repo
 
 git clone --bare https://gitlab.com/tezos/opam-repository.git upstream-opam-repository
-opam_repository_branch="$(git --git-dir=upstream-opam-repository name-rev --name-only "$opam_repository_tag" | cut -d'/' -f3)"
+opam_repository_branch="$(git --git-dir=upstream-opam-repository for-each-ref --contains "$opam_repository_tag" | grep "origin" | tail -n 1 | cut -d'/' -f4)"
```

Comment:
I suggest the single command with `git`:
```suggestion
opam_repository_branch="$(git --git-dir=upstream-opam-repository for-each-ref --contains "$opam_repository_tag" --count=1 --sort='-refname' --format="%(refname:lstrip=3)" refs/remotes/origin)"
```
it should do the same thing, but maybe also be a bit more robust (e.g. I think it works better if there is `origin` in a local branch name).

Please check it tho.